addons/html_builder/static/src/website_sale/products_item_option_plugin.js

```diff
@@ -0,0 +1,429 @@
+            name: "",
+            bg_color: "",
+            text_color: "",
+            position: "left",
+          };
+
+          return this._setRibbon(editingElement, ribbon, !isPreviewMode);
+        },
+      },
+      createRibbon: {
+        apply: ({ editingElement }) => {
+          this.productTemplateID = parseInt(
+            editingElement
+              .querySelector('[data-oe-model="product.template"]')
+              .getAttribute("data-oe-id")
+          );
+          const ribbonId = Date.now();
+          this.productTemplatesRibbons.push({
+            templateId: this.productTemplateID,
+            ribbonId: ribbonId,
+          });
+          const ribbon = reactive({
+            id: ribbonId,
+            name: "Ribbon Name",
+            bg_color: "",
+            text_color: "purple",
+            position: "left",
+          });
+          this.ribbons.push(ribbon);
+          this.ribbonsObject[ribbonId] = ribbon;
+          return this._setRibbon(editingElement, ribbon);
+        },
+      },
+      modifyRibbon: {
+        getValue: ({ editingElement, param }) => {
+          const field = param.mainParam;
+          let ribbonId = parseInt(editingElement.dataset.ribbonId);
+          if (!ribbonId) return;
+
+          return this.ribbonsObject[ribbonId][field];
+        },
+        isApplied: ({ editingElement, param, value }) => {
+          const field = param.mainParam;
+          let ribbonId = parseInt(editingElement.dataset.ribbonId);
+          if (!ribbonId) return;
+          if (!this.ribbonsObject[ribbonId]) {
+            ribbonId = Object.keys(this.ribbonsObject).find(
+              (key) => this.ribbonsObject[key].id === ribbonId
+            );
+            editingElement.dataset.ribbonId = ribbonId;
+          }
+          return this.ribbonsObject[ribbonId][field] === value;
+        },
+        apply: ({ editingElement, param, value }) => {
+          const isPreviewMode = historyPlugin.getIsPreviewing();
+          const setting = param.mainParam;
+          const ribbonId = parseInt(editingElement.dataset.ribbonId);
+          this.ribbonsObject[ribbonId][setting] = value;
+
+          const ribbon = this.ribbons.find((ribbon) => ribbon.id == ribbonId);
+          ribbon[setting] = value;
+
+          return this._setRibbon(editingElement, ribbon, !isPreviewMode);
+        },
+      },
+      deleteRibbon: {
+        apply: async ({ editingElement }) => {
+          const save = await new Promise((resolve) => {
+            this.services.dialog.add(ConfirmationDialog, {
+              body: _t("Are you sure you want to delete this ribbon?"),
+              confirm: () => resolve(true),
+              cancel: () => resolve(false),
+            });
+          });
+          if (!save) {
+            return;
+          }
+          return this._deleteRibbon(editingElement);
+        },
+      },
+    };
+  }
+
+  async onComponentCreated() {
+    this.ribbons = await this.loadRibbons();
```

Comment:
this function is doing 2 async requests. you could do them in parallel with Promise.all